hydradx/model/amm/basilisk_amm.py

```diff
@@ -0,0 +1,152 @@
+import copy
+import math
+
+
+class BasiliskPoolState:
+    unique_ids = {''}
+
+    def __init__(self, tokens: dict[str: float], trade_fee: float = 0):
+        """
+        tokens should be in the form of:
+        {
+            token1: quantity,
+            token2: quantity
+        }
+        There should only be two.
+        """
+        self.trade_fee = trade_fee
+        self.liquidity = dict()
+        self.asset_list: list[str] = []
+        self.fail = ''
+
+        for token, quantity in tokens.items():
+            self.asset_list.append(token)
+            self.liquidity[token] = quantity
+
+        self.shares = self.liquidity[self.asset_list[0]]
+
+        self.unique_id = ''
+        i = 0
+        while self.unique_id in BasiliskPoolState.unique_ids:
+            self.unique_id = '-'.join(self.asset_list + [str(i)])
+            i += 1
+
+    def copy(self):
+        new_self = copy.deepcopy(self)
+        new_self.fail = ''
+        return new_self
+
+    @property
+    def invariant(self):
+        return math.prod(self.liquidity.values())
+
+    def __repr__(self):
+        return (
+            f'BasiliskPool\n'
+            f'trade fee: {self.trade_fee}\n'
+            f'tokens: (\n'
+        ) + ')\n(\n'.join(
+            [(
+                f'    {token}\n'
+                f'    quantity: {self.liquidity[token]}\n'
+                f'    weight: {self.liquidity[token] / sum(self.liquidity.values())}\n'
+            ) for token in self.asset_list]
+        ) + '\n)'
+
+
+def add_risk_liquidity(
```

Comment:
Maybe it should be add_liquidity in both. I think it would be cool to have the method names and signatures the same in each case for better modularity

Reply:
Agreed, it should be add_liquidity for both.